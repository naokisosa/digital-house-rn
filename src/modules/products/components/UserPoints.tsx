import { StyleSheet, View } from 'react-native';
import React from 'react';
import { BaseText } from '@components';
import { colors } from '@style';

interface Props {
  month: string;
  totalPoints: string;
}

const UserPoints = ({ month, totalPoints }: Props) => {
  return (
    <>
      <BaseText style={styles.title}>TUS PUNTOS</BaseText>
      <View style={styles.card}>
        <BaseText style={styles.month}>{month}</BaseText>
        <BaseText style={styles.totalPoints}>{totalPoints}</BaseText>
      </View>
    </>
  );
};

export default UserPoints;

const styles = StyleSheet.create({
  title: {
    color: colors.gray,
    fontWeight: '800',
    fontSize: 14,
    lineHeight: 19.12,
    marginVertical: 20,
  },
  card: {
    backgroundColor: colors.primary,
    borderRadius: 20,
    paddingHorizontal: 19,
    paddingVertical: 21,
    height: 143,
    marginHorizontal: 34,
  },
  month: {
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 21.86,
    color: colors.white,
  },
  totalPoints: {
    fontWeight: '800',
    fontSize: 32,
    lineHeight: 43.71,
    color: colors.white,
    textAlign: 'center',
    marginTop: 7,
  },
});
