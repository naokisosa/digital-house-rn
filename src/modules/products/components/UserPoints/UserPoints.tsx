import { StyleSheet, View } from 'react-native';
import React, { memo } from 'react';
import { BaseText } from '@components';
import { colors } from '@style';

interface Props {
  month: string;
  totalPoints?: number;
}

const UserPoints = ({ month, totalPoints = 0 }: Props) => {
  return (
    <>
      <BaseText style={styles.title}>TUS PUNTOS</BaseText>
      <View style={[styles.card, styles.shadow]}>
        <BaseText style={styles.month}>{month}</BaseText>
        <BaseText style={styles.totalPoints}>{`${totalPoints.toLocaleString(
          'en-US'
        )} pts`}</BaseText>
      </View>
    </>
  );
};

export default memo(UserPoints);

const styles = StyleSheet.create({
  title: {
    color: colors.gray,
    fontWeight: '800',
    fontSize: 14,
    lineHeight: 19.12,
    marginVertical: 20,
  },
  card: {
    backgroundColor: colors.primary,
    borderRadius: 20,
    paddingHorizontal: 19,
    paddingVertical: 21,
    height: 143,
    marginHorizontal: 34,
  },
  shadow: {
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 4,
    },
    shadowOpacity: 0.5,
    shadowRadius: 4,

    elevation: 19,
  },
  month: {
    fontWeight: '800',
    fontSize: 16,
    lineHeight: 21.86,
    color: colors.white,
  },
  totalPoints: {
    fontWeight: '800',
    fontSize: 32,
    lineHeight: 43.71,
    color: colors.white,
    textAlign: 'center',
    marginTop: 7,
  },
});
