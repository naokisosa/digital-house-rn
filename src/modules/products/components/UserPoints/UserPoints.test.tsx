import 'react-native';
import React from 'react';

import UserPoints from './UserPoints';
import renderer from 'react-test-renderer';

describe('Test UserPoints component', () => {
  it('renders correctly with loading prop', () => {
    const { toJSON } = renderer.create(<UserPoints isLoading={true} month={'Enero'} />);
    expect(toJSON()).toMatchSnapshot();
  });
  it('renders correctly with totalPoints prop', () => {
    const { toJSON } = renderer.create(<UserPoints totalPoints={10000} month={'Enero'} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
