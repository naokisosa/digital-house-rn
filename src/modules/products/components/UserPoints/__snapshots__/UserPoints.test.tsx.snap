// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test UserPoints component renders correctly with loading prop 1`] = `
[
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Avenir",
        },
        {
          "color": "#9B9898",
          "fontSize": 14,
          "fontWeight": "800",
          "lineHeight": 19.12,
          "marginVertical": 20,
        },
      ]
    }
  >
    TUS PUNTOS
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "#334FFA",
          "borderRadius": 20,
          "height": 143,
          "marginHorizontal": 34,
          "paddingHorizontal": 19,
          "paddingVertical": 21,
        },
        {
          "elevation": 19,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <ActivityIndicator />
    </View>
  </View>,
]
`;

exports[`Test UserPoints component renders correctly with totalPoints prop 1`] = `
[
  <Text
    style={
      [
        {
          "color": "#000000",
          "fontFamily": "Avenir",
        },
        {
          "color": "#9B9898",
          "fontSize": 14,
          "fontWeight": "800",
          "lineHeight": 19.12,
          "marginVertical": 20,
        },
      ]
    }
  >
    TUS PUNTOS
  </Text>,
  <View
    style={
      [
        {
          "backgroundColor": "#334FFA",
          "borderRadius": 20,
          "height": 143,
          "marginHorizontal": 34,
          "paddingHorizontal": 19,
          "paddingVertical": 21,
        },
        {
          "elevation": 19,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 4,
            "width": 0,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 4,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "800",
            "lineHeight": 21.86,
          },
        ]
      }
    >
      Enero
    </Text>
    <Text
      style={
        [
          {
            "color": "#000000",
            "fontFamily": "Avenir",
          },
          {
            "color": "#FFFFFF",
            "fontSize": 32,
            "fontWeight": "800",
            "lineHeight": 43.71,
            "marginTop": 7,
            "textAlign": "center",
          },
        ]
      }
    >
      10,000 pts
    </Text>
  </View>,
]
`;
